File: app/src/main/java/com/example/belema/swiftkampus/CourseRegAdapter.java
package com.example.belema.swiftkampus;

import android.content.Context;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;

import com.example.belema.swiftkampus.gson.Course;

import java.util.List;

/**
 * Created by belema on 9/28/17.
 */

public class CourseRegAdapter extends ArrayAdapter<Course>{

    public CourseRegAdapter(@NonNull Context context, @LayoutRes int resource, List<Course> courseList) {
        super(context, resource, courseList);
    }

    @NonNull
    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {

        if (convertView == null) {
            convertView = LayoutInflater.from(parent.getContext())
                    .inflate(R.layout.course_register_layout, parent, false);
        }

        TextView courseCodeTextview = convertView.findViewById(R.id.tv_course_code);
        TextView courseNameTextView = convertView.findViewById(R.id.tv_course_name);

        Course course = getItem(position);

        courseCodeTextview.setText(String.valueOf(course.getCredit()));
        courseNameTextView.setText(course.getCourseName());

        return convertView;
    }
}

File: app/src/main/java/com/example/belema/swiftkampus/adapters/CourseRegAdapter.kt
package com.example.belema.swiftkampus.adapters

import android.content.Context
import android.support.annotation.LayoutRes
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.TextView

import com.example.belema.swiftkampus.R
import com.example.belema.swiftkampus.gson.Course

/**
 * Created by belema on 9/28/17.
 */

class CourseRegAdapter(context: Context, @LayoutRes resource: Int, courseList: List<Course>) : ArrayAdapter<Course>(context, resource, courseList) {

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View? {
        var view = convertView

        if (view == null) {
            view = LayoutInflater.from(parent.context)
                    .inflate(R.layout.course_register_layout, parent, false)
            }

        if (view != null) {
            val courseCodeTextView = view.findViewById<TextView>(R.id.tv_course_code)
            val courseNameTextView = view.findViewById<TextView>(R.id.tv_course_name)

            val course = getItem(position)
            if (course != null) {
                courseCodeTextView?.text = course.credit.toString()
                courseNameTextView?.text = course.courseName
            }
        }
        return view
    }
}
